test/test_linalg.py

```diff
@@ -6355,45 +6382,61 @@ def test__int4_mm(self, device, m, k, n):
 
         q_group = 32
         inner_k_tiles = 2
+        bias = torch.empty((0))
 
         torch.manual_seed(1)
         a_bf16 = torch.rand((m, k), dtype=torch.bfloat16, device=device)
         b_bf16 = torch.rand((k, n), dtype=torch.bfloat16, device=device)
 
         def convert_weight_to_int4pack(b):
-            b_uint8, b_scales_and_zeros = _group_quantize_tensor(
-                b, n_bit=4, q_group_size=q_group
-            )
+            if torch.backends.kleidiai.is_available() and device == "cpu":
+                b_fp32 = b.to(
+                    dtype=torch.float32
+                )  # Kleidiai only supports float32 datatype
+                b_uint8, b_scales_and_zeros = _group_quantize_tensor_symmetric(
+                    b_fp32, n_bit=4, groupsize=q_group, scheme="symmetric_groupwise"
+                )
```

Comment:
Meta folks,

Looking at this I am wondering, does it make sense for the Kleidi kernels to live in TorchAO? Seems like these kernels are tied to a specific quantization schema defined in TorchAO.

We might not be all set up to accept this in TorchAO, and there is already a precedence of doing this way in ATen but I am curious whether this is the right path long term?

CC: @malfet, @kimishpatel

Reply:
We had a discussion with @malfet on the complexities. We have a torchao patch in progress and can import the quantization scheme directly from torchao instead of adding _group_quantize_tensor_symmetric() in pytorch.

1. Integrating this change creates a circular dependency, we need to get this change merged first in order to upstream torchao with further changes. The idea is to make this PR self sufficient for aarch64 CI test cases and migrate the quantization schemes to torchao while integrating arm int4 kernels in existing int4_mm op.
2. Do we have source built torchao packages installed on the default CI test environment? is it ideal to import torchao in test_linalg.py?